src/cross_section/src/cross_section.cpp

```diff
@@ -132,6 +132,58 @@ void decompose_hole(const C2::PolyTreeD* outline,
   }
 }
 
+bool v2_lesser(glm::vec2 a, glm::vec2 b) {
+  if (a.x == b.x) return a.y < b.y;
+  return a.x < b.x;
+}
+
+double cross(glm::vec2 a, glm::vec2 b) { return a.x * b.y - a.y * b.x; }
+
+double distance(glm::vec2 a, glm::vec2 b) {
+  auto d = a - b;
+  return std::sqrt(d.x * d.x + d.y * d.y);
+}
+
+bool is_cw(glm::vec2 a, glm::vec2 b, glm::vec2 c) {
+  double lhs = cross(a - c, b - c);
+  double rhs = 1e-9 * distance(a, c) * distance(b, c);
```

Comment:
Can you skip the `sqrt`s by squaring both sides?

Reply:
Ahh good idea.

Reply:
I tried and didn't get it to work, not sure if I'm missing something. I've switch to `glm::distance` anyway, which works and lets me drop the helper.

Reply:
Maybe because you lose the sign of the cross? Could try `cross * abs(cross)`.

Reply:
Right, of course. Tests are passing with this:
```cpp
double SquaredDistance(glm::vec2 a, glm::vec2 b) {
  auto d = a - b;
  return d.x * d.x + d.y * d.y;
}

bool IsCw(glm::vec2 a, glm::vec2 b, glm::vec2 c) {
  double lhs = Cross(a - c, b - c);
  double rhs = 1e-18 * SquaredDistance(a, c) * SquaredDistance(b, c);
  return lhs * std::abs(lhs) <= rhs;
}
```